app/services/discourse_reactions/reaction_manager.rb

```diff
@@ -50,7 +54,17 @@ def reaction_scope
 
     def reaction_user_scope
       return nil unless @reaction
-      DiscourseReactions::ReactionUser.where(reaction_id: @reaction.id, user_id: @user.id)
+      search_reaction_user = DiscourseReactions::ReactionUser.where(user_id: @user.id, post_id: @post.id)
+      create_reaction_user = DiscourseReactions::ReactionUser.where(reaction_id: @reaction.id, user_id: @user.id, post_id: @post.id)
+      search_reaction_user.length > 0 ? search_reaction_user : create_reaction_user
```

Comment:
do we do the create query everytime even if sometimes we dont need it here?

Reply:
yes, we always use`first_or_initialize` method with this method

Reply:
What I mean is that create_reaction_user query will get executed each time, even if `search_reaction_user.length` > 0 given we call it before, we should avoid it, it's unnecessary work

Reply:
Yes that's right, ill try doing it in other way. As when we create the user we need `reaction_id` to create user successfully